crates/rpc/rpc/src/trace.rs

```diff
@@ -111,6 +116,37 @@ where
         })
     }
 
+    /// Performs multiple call traces on top of the same block. i.e. transaction n will be executed
+    /// on top of a pending block with all n-1 transactions applied (traced) first.
+    ///
+    /// Note: Allows to trace dependent transactions, hence all transactions are traced in sequence
+    pub async fn trace_call_many(
+        &self,
+        calls: Vec<(CallRequest, HashSet<TraceType>)>,
+        block_id: Option<BlockId>,
+    ) -> EthResult<Vec<TraceResults>> {
+        let at = block_id.unwrap_or(BlockId::Number(BlockNumberOrTag::Pending));
+        let (cfg, block_env, at) = self.eth_api.evm_env_at(at).await?;
```

Comment:
this will still error on `None`, so should `Pending` be the default here?
https://github.com/paradigmxyz/reth/blob/89c525f9c9399f699a19af3e6bf59ecd237599b3/crates/rpc/rpc/src/eth/api/transactions.rs#L145-L152

If `Pending` is the correct default, then it should be fine as long as we implement it later

Reply:
yes pending should be the default because this is supposed to operate on the very next state